interp/notation.ml

```diff
@@ -114,10 +114,22 @@ let level_eq_gen strict (l1, t1, u1) (l2, t2, u2) =
 
 let level_eq = level_eq_gen false
 
+let warn_undeclared_scope =
+  CWarnings.create ~name:"undeclared-scope" ~category:"parsing"
+                   (fun (scope) ->
+                    strbrk "Declaring a scope implicitly is deprecated; use in advance an explicit "
+                    ++ str "\"Declare Scope " ++ str scope ++ str ".\".")
+
 let declare_scope scope =
   try let _ = String.Map.find scope !scope_map in ()
   with Not_found ->
-(*    Flags.if_warn message ("Creating scope "^scope);*)
+    scope_map := String.Map.add scope empty_scope !scope_map
+
+(* To be removed after deprecation phase *)
+let declare_scope_if_needed scope =
```

Comment:
This looks like copy paste, I think `declare_scope` should take a flag to eventually raise the warning.

Reply:
I tried to use a flag but I'm not really satisfied by what I got. Tell me which version you prefer.